src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionMultilineStringArgsInspection.kt

```diff
@@ -16,20 +16,26 @@ class SmkSectionMultilineStringArgsInspection : SnakemakeInspection() {
             session: LocalInspectionToolSession
     ) = object : SnakemakeInspectionVisitor(holder, session) {
 
+        override fun visitSmkSubworkflowArgsSection(st: SmkSubworkflowArgsSection) {
+            checkArgumentList(st.argumentList, st)
+        }
+
         override fun visitSmkRuleOrCheckpointArgsSection(st: SmkRuleOrCheckpointArgsSection) {
-            if (st.sectionKeyword in setOf(SECTION_INPUT, SECTION_OUTPUT, SECTION_LOG)) {
                 checkArgumentList(st.argumentList, st)
-            }
         }
 
         private fun checkArgumentList(
                 argumentList: PyArgumentList?,
-                section: SmkRuleOrCheckpointArgsSection
+                section: SmkArgsSection
         ) {
 
             val args = argumentList?.arguments ?: emptyArray()
             args.forEach { arg ->
-                if (arg is PyStringLiteralExpression && arg.decodedFragments.size > 1) {
+                if (arg is PyStringLiteralExpression && arg.decodedFragments.size > 1
+                        && arg.stringElements.any { x ->
+                            x.nextSibling is PsiWhiteSpace
+                                    && x.nextSibling.textContains('\n')
+                        }) {
```

Comment:
Good, by one case isn't covered at all - when argument is string concatenation via '+', I've added and example, see a new failing test.

As one of the possible implementations, you could create a second visitor (e.g. PyElementVisitor) which traverse argument and visits only PyStringLiteralExpression and recursively handles PyBinaryExpression.

Reply:
Should I make helper visitor class private? It seems unlikely that someone can reuse it.

Reply:
No much sense to make it private